src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1746,3 +1748,25 @@ def set_attention_slice(self, slice_size: Optional[int]):
 
         for module in modules:
             module.set_attention_slice(slice_size)
+
+    def load_workflow(self, workflow_id, workflow_filename=None):
+        workflow_filename = workflow_filename or WORKFLOW_NAME
+
+        if os.path.isdir(workflow_id):
+            workflow_filepath = os.path.join(workflow_id, workflow_filename)
+        else:
+            workflow_filepath = hf_hub_download(repo_id=workflow_id, filename=workflow_filename)
+        workflow = self._dict_from_json_file(workflow_filepath)
```

Comment:
```suggestion
        self.workflow = self._dict_from_json_file(workflow_filepath)
```
I'd just save the workflow as an attribute here no?